crates/shared/src/account_balances/simulation.rs

```diff
@@ -80,9 +88,174 @@ impl Balances {
         tracing::trace!(?query, ?amount, ?simulation, "simulated balances");
         Ok(simulation)
     }
+}
+
+struct ZkSyncBalancesSimulator {
+    balances: contracts::support::Balances,
+    settlement: contracts::GPv2Settlement,
+    state_overrides: StateOverrides,
+    vault_relayer: H160,
+    vault: H160,
+    web3: Web3,
+}
+
+impl ZkSyncBalancesSimulator {
+    pub fn new(web3: Web3, settlement: H160, vault_relayer: H160, vault: H160) -> Self {
+        // pick a random address for the helper contract to avoid accidental collisions
+        let balances_address = H160::random();
```

Comment:
I think it's a bad practice to use random addresses in our code. This way we may get weird, non reproduceable behavior on single calls.